src/main/java/graphql/servlet/internal/ApolloSubscriptionKeepAliveRunner.java

```diff
@@ -0,0 +1,59 @@
+package graphql.servlet.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.websocket.Session;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+class ApolloSubscriptionKeepAliveRunner {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ApolloSubscriptionKeepAliveRunner.class);
+
+    private static final long KEEP_ALIVE_INTERVAL_SEC = 15;
+    private static final int EXECUTOR_POOL_SIZE = 10;
+
+    private ScheduledExecutorService executor;
+    private SubscriptionSender sender;
+    private ApolloSubscriptionProtocolHandler.OperationMessage keepAliveMessage;
+    private Map<Session, Future<?>> futures;
+
+    ApolloSubscriptionKeepAliveRunner(SubscriptionSender sender) {
+        this.sender = sender;
+        this.keepAliveMessage = ApolloSubscriptionProtocolHandler.OperationMessage.newKeepAliveMessage();
+        this.executor = Executors.newScheduledThreadPool(EXECUTOR_POOL_SIZE);
```

Comment:
Pass in the executor via the constructor so we can share the thread pool across connections (since this is created per-connection).

Reply:
(no longer applicable now that the runner is shared across all connections)